src/main/java/bio/terra/service/snapshotbuilder/SnapshotBuilderService.java

```diff
@@ -131,9 +134,21 @@ public SnapshotBuilderGetConceptsResponse searchConcepts(
       UUID datasetId, String domainId, String searchText, AuthenticatedUserRequest userRequest) {
     Dataset dataset = datasetService.retrieve(datasetId);
     TableNameGenerator tableNameGenerator = getTableNameGenerator(dataset, userRequest);
+    SnapshotBuilderSettings snapshotBuilderSettings =
+        snapshotBuilderSettingsDao.getSnapshotBuilderSettingsByDatasetId(datasetId);
+
+    SnapshotBuilderDomainOption snapshotBuilderDomainOption =
+        snapshotBuilderSettings.getDomainOptions().stream()
+            .filter(domainOption -> Objects.equals(domainOption.getCategory(), domainId))
+            .findFirst()
+            .orElseThrow(
+                () ->
+                    new BadRequestException(
+                        String.format("Invalid domain category is given: %s", domainId)));
```

Comment:
In a case like this I think `formatted()` is slightly better
```suggestion
                       "Invalid domain category is given: %s".formatted(domainId))));
```

Or you could use concatenation
```suggestion
                       "Invalid domain category is given: " + domainId)));
```